pkg/cache/context.go

```diff
@@ -271,13 +284,7 @@ func (ctx *Context) updatePodInCache(oldObj, newObj interface{}) {
 func (ctx *Context) filterPods(obj interface{}) bool {
 	switch obj := obj.(type) {
 	case *v1.Pod:
-		// if a terminated pod is added to cache, it will
-		// add requested resource to the cached node, causing
-		// the node uses more resources that it actually is,
-		// this can only be fixed after the pod is removed.
-		// (trigger the delete pod)
-		return utils.GeneralPodFilter(obj) &&
-			!utils.IsPodTerminated(obj)
+		return utils.GeneralPodFilter(obj)
```

Comment:
This logic gets moved to `addPodToCache()` / `updatePodInCache()` as we DO want to allow terminated pods to flow through to `removePodFromCache()`. The old logic filtered these out from all methods, resulting in leaks of terminated pods.